### ISSUE TITLE
User friendly name of Downloaded Templates  Volumes and ISOs

### ISSUE THREAD

**Issue by alexandremattioli** [apache/cloudstack#6949]
##### ISSUE TYPE

 * Enhancement Request

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.17.1
~~~

##### OS / ENVIRONMENT

N/A

##### STEPS TO REPRODUCE

~~~
End user clicks on Download Template, ISO or Volume
~~~

##### EXPECTED RESULTS

~~~
Filename in the format "resourceName.ext" for templates, isos or volumes being downloaded.
~~~

##### ACTUAL RESULTS

~~~
Filename in the format "UUID.ext" for templates, isos or volumes being downloaded.
~~~

**Comment by nvazquez** [comment]
@alexandremattioli I've added to the GSOC2023 ideas, can you please present the idea in a paragraph for better understanding of potential contributors?

**Comment by alexandremattioli** [comment]
@nvazquez 
Currently in ACS when an end-user clicks on the download button for a tempalte, iso or volume, the file name is in the form UUID.ext.  It would be extremely useful for the file name to be resourceName.ext instead.

**Comment by DaanHoogland** [comment]
> @nvazquez Currently in ACS when an end-user clicks on the download button for a tempalte, iso or volume, the file name is in the form UUID.ext. It would be extremely useful for the file name to be resourceName.ext instead.

@alexandremattioli, I think what @nvazquez means is to edit the issue description to read as a full functional spec, so it will be easy to assess for interns if they want to work on it.

**Comment by alexandremattioli** [comment]
Copy @DaanHoogland

**Comment by nvazquez** [comment]
Yes, @DaanHoogland is correct, sorry if it wasn't clear

**Comment by nvazquez** [comment]
Hi @AayushSaini101,

Thanks for reaching us, I would recommend a good way to start is the Hackerbook: https://github.com/shapeblue/hackerbook, and for further reading on the core concepts, the documentation: http://docs.cloudstack.apache.org/en/latest/

Please let us know if you have any issues

**Comment by AayushSaini101** [comment]
Thanks, @nvazquez  for the reply. Over the last two days. I am trying to set up the project, but unable to execute the server management.  I am getting build failure error and cannot able to execute the server. I am using rhel8 and project  SDK is 8 and also I have tried by using setting SDk to 11.
![image](https://user-images.githubusercontent.com/60972989/228311897-5c3533d8-b0ed-40a8-9c14-db1d97de67c6.png)

**Comment by nvazquez** [comment]
@AayushSaini101 can you pull the latest code form the main branch? Otherwise, you can try pulling the code from the latest release tag: 4.18.0.0. Also, it seems like you are building using the noredist profile, is this correct? If so, please get the dependencies from this repo: https://github.com/shapeblue/cloudstack-nonoss into your `deps` directory and install them via the script before building

**Comment by AayushSaini101** [comment]
@nvazquez Thanks above changes works. Now I am facing an issue regarding the DB Setup while executing the mvn -P developer -pl developer -Ddeploydb

![image](https://user-images.githubusercontent.com/60972989/228510478-143fbb69-1ab4-45c4-b48a-ca721add13db.png)

**Comment by AayushSaini101** [comment]
@shwstppr Thanks, I am trying this. Also replacing MariaDB to Mysql in my local

**Comment by AayushSaini101** [comment]
@shwstppr @nvazquez Still I cannot able to run it is showing 
![image](https://user-images.githubusercontent.com/60972989/228815593-e221d381-657f-420b-81a7-5a81d373804d.png)
I have already followed the above instructions in the handbook. I am using MariaDB

**Comment by AayushSaini101** [comment]
![image](https://user-images.githubusercontent.com/60972989/228821083-76498b21-e5ae-4afd-95ef-a10b1858f242.png)
Thanks but still is showing Access Denied. Granting to root@localhost
![image](https://user-images.githubusercontent.com/60972989/228821261-6051fb9f-30d8-41e8-a572-1581bd6fa47a.png)

**Comment by AayushSaini101** [comment]
@nvazquez Thanks But it contains only one file db.properties   
![image](https://user-images.githubusercontent.com/60972989/228833643-52320a99-90a1-4f20-9c28-59059b461307.png)

**Comment by nvazquez** [comment]
@AayushSaini101 ok, are you able to access mysql using the password set on the file by `mysql -u root -p`?

**Comment by AayushSaini101** [comment]
@nvazquez  Yes 

![image](https://user-images.githubusercontent.com/60972989/228840812-7e091121-5b87-47d2-b35c-af49a65f680c.png)

**Comment by nvazquez** [comment]
@AayushSaini101 can you post the content of the db.properties file hiding the passwords? I suspect the root password field may be empty

**Comment by AayushSaini101** [comment]
@nvazquez

**Comment by nvazquez** [comment]
@AayushSaini101 thanks for sharing, can you change these first lines to:

````
db.cloud.username=cloud
db.cloud.password=cloud
db.root.password=<YOUR_ROOT_PASSWORD>
````

CloudStack will create the cloud user (you can choose another password for it) via root user

**Comment by AayushSaini101** [comment]
Thanks @nvazquez Finally Resolved. How to switch this 
![image](https://user-images.githubusercontent.com/60972989/228864799-f96aae1d-84d6-4a20-9892-7687efa46e29.png)

**Comment by nvazquez** [comment]
@AayushSaini101 great! Now you only need to bring up the UI to interact with the management server. Please check: https://github.com/shapeblue/hackerbook/blob/main/2-dev.md#building-cloudstack last section:

````
To bring up the Cloudstack Ui

    $ cd /path/to/cloudstack/ui
    $ sudo apt install nodejs npm
    $ npm install
    $ npm run serve

Make sure to set CS_URL=http://localhost:8080/client on .env.local file on ui.

You should be able to run the management server on http://localhost:5050/
````

**Comment by AayushSaini101** [comment]
Thanks working on it

**Comment by nvazquez** [comment]
Great, well done @AayushSaini101 - let us know if you have further issues or questions

**Comment by AayushSaini101** [comment]
@nvazquez  I am unable to add Register ISO, it is showing unable to find image store. I tried to find this in docs, but able to find. 

![image](https://user-images.githubusercontent.com/60972989/229103634-54888131-8139-4de8-9ba4-8dae25f0b3f8.png)

**Comment by nvazquez** [comment]
@AayushSaini101 looks like you don’t have a secondary storage defined, which hypervisor are you using? You can consider using the simulator that allows you to create a test simulated zone from a script

**Comment by AayushSaini101** [comment]
@nvazquez Can you please tell me where script is located also can you please refer to some of the videos to understand cloudstack much better. In Installation guide there are many videos which are outdated

**Comment by nvazquez** [comment]
@AayushSaini101 you may need to build the project and setup databases for simulator as explained on: https://cwiki.apache.org/confluence/plugins/servlet/mobile?contentId=30748425#content/view/30748425 

After that, with your management server running, you can deploy a test zone with this command:

````
python ./tools/marvin/marvin/deployDataCenter.py -i setup/dev/advanced.cfg
````

(may need to run this before: pip install tools/marvin/dist/Marvin-*.tar.gz)

**Comment by DaanHoogland** [comment]
fixed in #9252
